lib/src/main/kotlin/com/lemonappdev/konsist/core/declaration/KoDeclaration.kt

```diff
@@ -30,35 +31,24 @@ abstract class KoDeclaration(private val ktTypeParameterListOwner: KtTypeParamet
         .annotationEntries
         .map { KoAnnotation.getInstance(it) }
 
-    fun hasPublicModifier() = ktTypeParameterListOwner.hasModifier(KtTokens.PUBLIC_KEYWORD)
+    val modifiers by lazy {
+        ktTypeParameterListOwner
+            .modifierList
+            ?.text
+            ?.split(" ")
+            ?.map { KoModifier.valueOf(it.uppercase()) }
+            ?: emptyList()
+    }
 
-    fun isPublicOrDefault() = ktTypeParameterListOwner.run {
-        if (hasModifier(KtTokens.PUBLIC_KEYWORD)) {
-            return@run true
-        }
+    fun hasPublicModifier() = hasModifiers(KoModifier.PUBLIC)
 
-        val hasOtherVisibilityModifier =
-            hasModifier(KtTokens.PRIVATE_KEYWORD) ||
-                hasModifier(KtTokens.PROTECTED_KEYWORD) ||
-                hasModifier(KtTokens.INTERNAL_KEYWORD)
+    fun isPublicOrDefault() = ktTypeParameterListOwner.isPublic
```

Comment:
Let's use hasModifiers(KoModifier.X) for every modifier property because "kt methods" may have additional behavior